From eb4441b022334c0daf46e2c99601228efee64c39 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Wed, 22 Aug 2018 16:44:21 +0800
Subject: [PATCH 4545/5242] MLK-19442-2 phy: phy-fsl-imx8mq-usb: change
 ssc_range value

commit  a48a65a40113b9b5d40114d02a5877d089f523a9 from
https://source.codeaurora.org/external/imx/linux-imx.git

According to IC engineer suggestion, set ssc_range as -4003 ppm
will have more tolerence for EMI, and suitable for more boards.
Besides, one customer board needs to set this value to pass TX
SSC test.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/phy/phy-fsl-imx8mq-usb.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/phy/phy-fsl-imx8mq-usb.c b/drivers/phy/phy-fsl-imx8mq-usb.c
index 113bc5b..3705546 100644
--- a/drivers/phy/phy-fsl-imx8mq-usb.c
+++ b/drivers/phy/phy-fsl-imx8mq-usb.c
@@ -15,6 +15,8 @@
 
 #define PHY_CTRL0			0x0
 #define PHY_CTRL0_REF_SSP_EN		BIT(2)
+#define PHY_CTRL0_SSC_RANGE_MASK	(7 << 21)
+#define PHY_CTRL0_SSC_RANGE_4003PPM	(0x2 << 21)
 
 #define PHY_CTRL1			0x4
 #define PHY_CTRL1_RESET			BIT(0)
@@ -66,6 +68,8 @@ static void imx8mq_usb_phy_init(struct imx8mq_usb_phy *phy)
 
 	value = readl(phy->base + PHY_CTRL0);
 	value |= PHY_CTRL0_REF_SSP_EN;
+	value &= ~PHY_CTRL0_SSC_RANGE_MASK;
+	value |= PHY_CTRL0_SSC_RANGE_4003PPM;
 	writel(value, phy->base + PHY_CTRL0);
 
 	value = readl(phy->base + PHY_CTRL2);
-- 
1.7.9.5

